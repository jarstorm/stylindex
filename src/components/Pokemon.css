.pokemon-avatar {
	height: 40px;
	float: right;
	margin-top: -10px;
}

.more-info-link {
	float: right;
}