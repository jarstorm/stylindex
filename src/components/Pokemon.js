import React, {Component} from 'react';
import EvolutionList from './EvolutionList';

import './Pokemon.css';

export default class Pokemon extends Component {
  render() {
    const {data} = this.props;
    return(
      <div className={this.props.extraClass}>
        <div className="panel panel-default">
          <div className="panel-heading">
            <h3>{data.name}<img className="pokemon-avatar" src={data.image} alt="pokemon avatar"/></h3>  
          </div>
          <div className="panel-body">
            <p>Id: {data.id}</p>
            <p>Number: {data.number}</p>            
            <p>Max HP: {data.maxHP}</p>
            <p>Max CP: {data.maxCP}</p>
            <p>Types: {data.types}</p>
            <EvolutionList data={data} />
          </div>
        </div>
      </div> 
    )   
  }
}
