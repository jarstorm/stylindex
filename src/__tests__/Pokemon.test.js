// Link.react.test.js
import React from 'react';
import Pokemon from '../components/Pokemon';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('Draw Insights component', () => {  
  const data = { "id": "UG9rZW1vbjowMDE=",
      "number": "001",
      "name": "Bulbasaur",
      "maxCP": 951,
      "maxHP": 1071,
      "image": "https://img.pokemondb.net/artwork/bulbasaur.jpg",
      "types": [
        "Grass",
        "Poison"
      ]};
  const component = renderer.create(<Pokemon data={data} />);
  let json = component.toJSON();
  expect(json).toMatchSnapshot();  
});