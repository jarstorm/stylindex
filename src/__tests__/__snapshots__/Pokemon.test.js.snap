// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draw Insights component 1`] = `
<div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3>
        Bulbasaur
        <img
          className="pokemon-avatar"
          src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
        />
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <p>
        Id: 
        UG9rZW1vbjowMDE=
      </p>
      <p>
        Number: 
        001
      </p>
      <p>
        Max HP: 
        1071
      </p>
      <p>
        Max CP: 
        951
      </p>
      <p>
        Types: 
        Grass
        Poison
      </p>
      <div />
    </div>
  </div>
</div>
`;
